src/gpu/nvidia/sycl_cuda_compat.hpp

```diff
@@ -35,11 +35,16 @@ T get_native_mem(const interop_handle &ih, U acc) {
             ih.get_native_mem<::sycl::backend::ext_oneapi_cuda>(acc));
 }
 
-template <typename T>
-void host_task(::sycl::handler &cgh, const T &task) {
-    cgh.host_task(task);
+template <typename HandlerT, typename FnT>
+void host_task(HandlerT &cgh, const FnT &task) {
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle ih) {
+#else
+    cgh.host_task([=](sycl::interop_handle ih) {
+#endif
+        task(std::move(ih));
+    });
 }
-
```

Comment:
(nit) maybe restore the newline for readability?

Reply:
Line has been added back in.